src/PrestaShopBundle/Service/Cache/Refresh.php

```diff
@@ -50,51 +49,61 @@ public function __construct($env = 'prod')
     {
         umask(0000);
         set_time_limit(0);
-        $this->env = $env;
-        $this->output = new NullOutput();
+        $this->env = _PS_MODE_DEV_ ? 'dev' : 'prod';
         $this->commands = [];
 
         require_once _PS_ROOT_DIR_.'/app/AppKernel.php';
-        $kernel = new \AppKernel($this->env, false);
+        $kernel = new \AppKernel($this->env, _PS_MODE_DEV_);
         $this->application = new Application($kernel);
         $this->application->setAutoExit(false);
     }
 
     /**
-     * add cache clear
+     * add cache clear.
      */
     public function addCacheClear()
     {
         $this->commands[] = ['command' => 'cache:clear', '--env' => $this->env, '--no-debug' => true];
     }
 
     /**
-     * add doctrine schema update
+     * add doctrine schema update.
      */
     public function addDoctrineSchemaUpdate()
     {
         $this->commands[] = ['command' => 'doctrine:schema:update', '--env' => $this->env, '--no-debug' => false, '--force' => true];
     }
 
     /**
-     * Execute all defined commands
+     * Execute all defined commands.
      *
      * @throws \Exception if no command defined
      */
     public function execute()
     {
-        $output = [];
+        $output = null;
+
         if (empty($this->commands)) {
             throw new \Exception('Error, you need to define at least on command');
         }
 
         foreach ($this->commands as $command) {
-            $exitCode = $this->application->run(new ArrayInput($command), $this->output);
+            if (_PS_MODE_DEV_ && isset($command['--no-debug'])) {
+                $command['--no-debug'] = false;
+            }
+
+            if (false === PS_MODE_DEV_) {
+                $output = new NullOutput();
+            }
+
+            $exitCode = $this->application->run(new ArrayInput($command), $output);
+
+            $commandOutput = [];
```

Comment:
(I fell on this PR because it can fix one of my issues)

I agree with [Julien de Martin](https://github.com/Shudrum), we should return all command results.

I would have the following code to be generic:
```php
$commandOutput[$command['command']] = $exitCode;
```

Let me know if you change something, we will have to adapt the following code:
https://github.com/PrestaShop/PrestaShop/blob/7fd1b508b1eb24f4cc3fc95db82274ae9a96aa51/install-dev/models/install.php#L188